packages/shared/lib/transactionHistory.ts

```diff
@@ -0,0 +1,38 @@
+import type { WalletAccount } from './typings/wallet'
+
+export const generateTransactionHistoryCsvFromAccount = (WalletAccount: WalletAccount): string => {
+    const headers = 'id,timestamp,value,internal\r\n'
+    let csv = headers
+
+    WalletAccount.messages.forEach((message) => {
+        if (message.payload.type === 'Transaction') {
+            const id = message.id
+            const timestamp = message.timestamp
+            const incoming = message.payload.data.essence.data.incoming
+            const value = message.payload.data.essence.data.value
+            const internal = message.payload.data.essence.data.internal
```

Comment:
It might be cleaner to do this in two lines with object destructuring syntax:

```suggestion
            const { id, timestamp } = message
            const { incoming, value, internal } = message.payload.data.essence.data
```